types/fqdn.pp

```diff
@@ -1,2 +1,2 @@
 # @summary Validate a Fully Qualified Domain Name
-type Stdlib::Fqdn = Pattern[/\A(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\z/]
+type Stdlib::Fqdn = Pattern[/\A((([[:alnum:]]|[[:alnum:]][[:alnum:]-]*[[:alnum:]])\.)*[[:alpha:]]+)\z/]
```

Comment:
also changing the end of this to `*[[:alpha:]]+` is invalid and breaking.  There are two issues with this
 * A fqdn can contain just one label i.e. com, net, google are all valid FQDN's  as such this type can and will be used to validated hostnames which could contain a numbers e.g. webserver1001 
 * This only assumes the IANA DNS it doesn't account for:
   * private TLDs e.g. 7up may have there own private DNS in there compan
   * Pseudo TLD's e.g. [i2p](https://en.wikipedia.org/wiki/I2P)

Arguably the IANA DNS is the one most users would want to check however i think we should split that out to 

 Stdlib::DNS::IANA 

ill copy this response to #1282

Reply:
Reverted and adjusted the test suite accordingly.